src/client.ts

```diff
@@ -238,6 +243,21 @@ export default class Client implements ClientInterface {
     this.#ruby = ruby;
   }
 
+  async determineFormatter() {
+    const configuration = vscode.workspace.getConfiguration("rubyLsp");
+    const configuredFormatter = configuration.get("formatter")!;
+
+    if (configuredFormatter === "auto") {
+      if (await this.projectHasDependency(/^rubocop-?/)) {
+        this.formatter = "rubocop";
+      } else if (await this.projectHasDependency("syntax_tree")) {
+        this.formatter = "syntax_tree";
+      } else {
+        this.formatter = "none";
+      }
+    }
+  }
```

Comment:
After pairing with @andyw8, a change like this is needed to display the formatter when the configured value isn't `"auto"`

```suggestion
  async determineFormatter() {
    const configuration = vscode.workspace.getConfiguration("rubyLsp");
    const configuredFormatter: string = configuration.get("formatter")!;

    if (configuredFormatter === "auto") {
      if (await this.projectHasDependency(/^rubocop-?/)) {
        this.formatter = "rubocop";
      } else if (await this.projectHasDependency("syntax_tree")) {
        this.formatter = "syntax_tree";
      } else {
        this.formatter = "none";
      }
    } else if (configuredFormatter !== undefined) {
      this.formatter = configuredFormatter;
    }
  }
```